### ISSUE TITLE
Closest node is not always in the same room!

### ISSUE THREAD

**Issue by hawesie** [strands-project/strands_navigation#219]
I'm not sure there is an easy way to fix this, and I'm pretty sure we can work around it by setting influence zones correctly, but I'm posting this for the record and to make people setting up topological maps aware of the issue.

When the robot is not _at_ a topological node (i.e. not in its influence zone) it plans a route from the _closest_ node using (I assume) euclidean distance. This distance measure does not respect the graph or the underlying metric map thus you can have cases like the one below where the robot is not at a node and closest to WayPoint7. When asked to move to WayPoint6 it plans with movebase because that is the edge type from WayPoint7 to WayPoint6.

The lesson is that if you're working in constrained environments like this, be sure to set influence zones!

![screen shot 2015-04-10 at 22 46 51](https://cloud.githubusercontent.com/assets/927289/7097262/4a4f0fda-dfd4-11e4-9c30-b8a09c61c144.png)

**Comment by Jailander** [comment]
Yes, this definitely happens, at some point I thought of using navfn to plan to the three closest points and see the shortest route, but never tried it in the end, maybe I could do that, for now I would recommend setting influence zones that go up the walls in this case.
